src/Compilers/CSharp/Portable/Compilation/MethodBodySemanticModel.cs

```diff
@@ -103,6 +103,7 @@ internal override BoundNode Bind(Binder binder, CSharpSyntaxNode node, BindingDi
                 case SyntaxKind.RemoveAccessorDeclaration:
                 case SyntaxKind.CompilationUnit:
                 case SyntaxKind.RecordDeclaration:
+                case SyntaxKind.RecordStructDeclaration:
```

Comment:
>case SyntaxKind.RecordStructDeclaration: [](start = 16, length = 40)

Is this case reachable?

Reply:
I spent some time on this yesterday, but I was not able to reach either the Record or RecordStruct case. Didn't find a test  that breaks if I removed the Record case above.
I can add a PROTOTYPE. Let me know.

---
In reply to: [590813467](https://github.com/dotnet/roslyn/pull/51720#discussion_r590813467) [](ancestors = 590813467)

Reply:
> I spent some time on this yesterday, but I was not able to reach either the Record or RecordStruct case. Didn't find a test  that breaks if I removed the Record case above.
I can add a PROTOTYPE. Let me know.

I think first we need to answer a question whether a record struct primary constructor has any user defined body. What might that be? If none, then we couldn't even have a method body semantic model for it.

---
In reply to: [590816107](https://github.com/dotnet/roslyn/pull/51720#discussion_r590816107) [](ancestors = 590816107,590813467)